publisher.go

```diff
@@ -14,17 +14,15 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package kubewatcher
+package fission
 
-import (
-	"k8s.io/client-go/1.5/pkg/watch"
-)
+import "net/http"
 
 type (
 	Publisher interface {
-		// Publish an event to a "target".  Target's meaning depends on the
+		// Publish an request to a "target".  Target's meaning depends on the
 		// publisher: it's a URL in the case of a webhook publisher, or a queue
 		// name in a queue-based publisher such as NATS.
-		Publish(event watch.Event, target string)
+		Publish(req *http.Request)
```

Comment:
It'sn not a good idea to have an http request object in there when it may not always be published over http.  (I know the only implementation right now is http, but it won't always be the case.)

I don't understand the last part of your comment -- what do you mean by "restrict the protocol"?  Are you suggesting a different publisher interface?